dspy/clients/openai.py

```diff
@@ -0,0 +1,403 @@
+from collections import defaultdict
+import re
+import time
+from typing import Any, Dict, List, Optional, Union
+
+import openai
+
+from dspy.utils.logging import logger
+from dspy.clients.finetune import (
+    FinetuneJob,
+    TrainingMethod,
+    TrainingStatus,
+    validate_finetune_data,
+    save_data,
+)
+
+# Provider name
+PROVIDER_OPENAI = "openai"
+
+# List of model IDs
+_MODEL_IDS = [
```

Comment:
This is hard to maintain, as openai releases pretty quickly. We can do something like:
```
client = OpenAI()
models = client.models.list()
```
when we need to get the model list.